azurerm/resource_arm_stream_analytics_reference_input_blob.go

```diff
@@ -0,0 +1,243 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+
+	"github.com/Azure/azure-sdk-for-go/services/streamanalytics/mgmt/2016-03-01/streamanalytics"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmStreamAnalyticsReferenceInputBlob() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmStreamAnalyticsReferenceInputBlobCreateUpdate,
+		Read:   resourceArmStreamAnalyticsReferenceInputBlobRead,
+		Update: resourceArmStreamAnalyticsReferenceInputBlobCreateUpdate,
+		Delete: resourceArmStreamAnalyticsReferenceInputBlobDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"stream_analytics_job_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"date_format": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"path_pattern": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
Could we get a `validate.NoEmptyStrings` here?

Reply:
Done.

Reply:
Just curious, though. With `Required: true`, isn't `NoEmptyStrings` enforced automatically?

Reply:
Nope, you can pass an empty string, alls that is required is something is assinged to the property

Reply:
Ah gotcha. Would you like me to add `validate.NoEmptyStrings` here then?